lib/si-layer-cache/src/retry_queue.rs

```diff
@@ -348,65 +364,147 @@ impl RetryQueueManager {
-                );
+        Ok(())
+    }
 
-                gauge!(
-                    layer_cache_retry_queue_depth = pending_files.len(),
-                    cache_name = cache_name
-                );
+    /// Migrate old retry queue files to backend-specific directories
+    async fn migrate_old_queue_files(
+        &mut self,
+        old_dir: &std::path::Path,
+        cache_name: &str,
+        target_backend: BackendType,
+    ) -> LayerDbResult<()> {
+        use telemetry::prelude::*;
+        use tokio::fs;
 
-                let mut state = QueueState::new(retry_dir, self.config.initial_backoff);
-                state.pending_files = pending_files;
-                // Retry immediately on startup
-                state.next_retry_time = Instant::now();
+        let new_dir = self.retry_dir_for_backend(cache_name, target_backend);
 
-                self.queues.insert(cache_name.to_string(), state);
+        // Create new directory
+        fs::create_dir_all(&new_dir)
+            .await
+            .map_err(LayerDbError::RetryQueueDirCreate)?;
+
+        // Move all .pending files
+        let mut entries = fs::read_dir(old_dir)
+            .await
+            .map_err(LayerDbError::RetryQueueDirRead)?;
+
+        let mut migrated_count = 0;
+
+        while let Some(entry) = entries
+            .next_entry()
+            .await
+            .map_err(LayerDbError::RetryQueueDirRead)?
+        {
+            let old_path = entry.path();
+
+            if old_path.extension() == Some(std::ffi::OsStr::new("pending")) {
+                let file_name = entry.file_name();
+                let new_path = new_dir.join(&file_name);
+
+                // Move file atomically
+                fs::rename(&old_path, &new_path)
+                    .await
+                    .map_err(LayerDbError::RetryQueueFileWrite)?;
+
+                migrated_count += 1;
             }
         }
 
+        // Try to remove old directory (will fail if not empty, which is fine)
+        let _ = fs::remove_dir(old_dir).await;
+
+        info!(
+            cache.name = cache_name,
+            backend = ?target_backend,
+            migrated_files = migrated_count,
+            "migrated old retry queue files to new format"
+        );
+
         Ok(())
     }
 
     /// Enqueue a failed write for retry
-    pub async fn enqueue(&mut self, event: LayeredEvent) -> LayerDbResult<()> {
+    pub async fn enqueue(
+        &mut self,
+        event: LayeredEvent,
+        backend: BackendType,
+    ) -> LayerDbResult<()> {
         use telemetry::prelude::*;
         use tokio::fs;
         use ulid::Ulid;
 
         let cache_name = event.payload.db_name.to_string();
-        let retry_dir = self.config.base_path.join(format!("{cache_name}_retries"));
+        let retry_dir = self.retry_dir_for_backend(&cache_name, backend);
 
         // Ensure retry directory exists
         fs::create_dir_all(&retry_dir)
```

Comment:
I know this is in the prior version, but could this call be from `std`, i.e. the blocking version?